ReactAndroid/src/main/java/com/facebook/react/devsupport/DebugOverlayController.java

```diff
@@ -96,25 +97,51 @@ public DebugOverlayController(ReactContext reactContext) {
     mWindowManager = (WindowManager) reactContext.getSystemService(Context.WINDOW_SERVICE);
   }
 
-  public void setFpsDebugViewVisible(boolean fpsDebugViewVisible) {
-    if (fpsDebugViewVisible && mFPSDebugViewContainer == null) {
-      if (!permissionCheck(mReactContext)) {
-        FLog.d(ReactConstants.TAG, "Wait for overlay permission to be set");
-        return;
-      }
-      mFPSDebugViewContainer = new FpsView(mReactContext);
-      WindowManager.LayoutParams params = new WindowManager.LayoutParams(
+  private void showFpsDebugView() {
+    UiThreadUtil.runOnUiThread(new Runnable() {
+      @Override
+      public void run() {
+        if (!permissionCheck(mReactContext)) {
+          FLog.d(ReactConstants.TAG, "Wait for overlay permission to be set");
+          return;
+        }
+
+        if (mFPSDebugViewContainer != null) {
```

Comment:
@bae-unidev you might want to swap that with the `permissionCheck` as this null check should run faster.

Reply:
Thanks, I will fix it.